src/physics/mod.rs

```diff
@@ -4,12 +4,148 @@ pub mod rapier;
 #[cfg(feature = "avian")]
 pub mod avian;
 
-use bevy::prelude::*;
+use std::sync::Arc;
+
+use bevy::{ecs::system::EntityCommands, prelude::*};
 use bevy_ecs_tilemap::prelude::*;
 use tiled::{ObjectData, ObjectLayerData};
 
 use crate::prelude::*;
 
+/// Trait defining a generic way of handling colliders
+/// across different physics backends.
+pub trait HandleColliders {
+    /// Load shapes from an object layer as physics colliders.
+    ///
+    /// By default `bevy_ecs_tiled` will only process object layers
+    /// named in `collision_layer_names` in `TiledMapSettings`,
+    /// and tileset collision shapes named in `collision_object_names`.
+    ///
+    /// Collision layer names are case-insensitive and leading/trailing
+    /// whitespace is stripped out.
+    fn insert_colliders_from_shapes(
+        &self,
+        parent_entity: Entity,
+        map_type: &TilemapType,
+        grid_size: Option<&TilemapGridSize>,
+        object_data: &ObjectData,
+    ) -> EntityCommands;
+}
+
+#[derive(Clone, Resource)]
+pub enum PhysicsBackend {
+    Rapier,
+    Avian,
+    Custom(Arc<Box<dyn HandleColliders + Send + Sync>>),
+}
+
+impl PhysicsBackend {
+    pub fn insert_object_colliders(
+        &self,
+        commands: &mut Commands,
+        object_entity: Entity,
+        map_type: &TilemapType,
+        object_data: &ObjectData,
+        collider_callback: ColliderCallback,
+    ) {
+        match self {
+            #[cfg(feature = "rapier")]
+            PhysicsBackend::Rapier => {
+                let mut e = rapier::insert_rapier_colliders_from_shapes(
+                    commands,
+                    object_entity,
+                    map_type,
+                    None,
+                    object_data,
+                    collider_callback,
+                );
+
+                collider_callback(&mut e);
+            }
+            #[cfg(feature = "avian")]
+            PhysicsBackend::Avian => {
+                let e = avian::insert_avian_colliders_from_shapes(
+                    commands,
+                    object_entity,
+                    map_type,
+                    None,
+                    object_data,
+                );
+
+                let mut e = e.expect("failure creating entity for Avian collider");
+
+                collider_callback(&mut e);
+            }
+            PhysicsBackend::Custom(backend) => {
+                let mut e = backend.insert_colliders_from_shapes(
+                    object_entity,
+                    map_type,
+                    None,
+                    object_data,
+                );
+
+                collider_callback(&mut e);
+            }
+            _ => panic!("avian or rapier physics backend supplied but feature is disabled"),
```

Comment:
Instead of this global "catch all" case, it could be nice to have something more specific.
For instance, for avian :

```rust
PhysicsBackend::Avian => {
  #[cfg(feature = "avian")] {
    // do stuff
  }
  #[cfg(not(feature = "avian"))] {
    panic!("Requested Avian physics backend but feature is disabled");
  }
}
```